### PR TITLE
Fix missing tkr annotation on AWS prod cluster package based lcm

### LINKED ISSUE #2863: ClusterClass based AWS Clusters fail to provision with prod plan
**Bug description**
Creating an AWS cluster based on the new aws clusterclass fails when trying to create a `prod` cluster. Error is trying to resolve TKRs for md-1 and md-2

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [x] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
Expect to successfully create an AWS prod cluster

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by tenczar**
Updates the AWS overlay for classy clusters. This adds the TKR resolve
annotation to the second and third machine deployments.

### What this PR does / why we need it
For package based lcm clusters we need to add the tkr-resolve annotation to additional machine deployments that are added for `prod` plan clusters.
### Which issue(s) this PR fixes

Fixes #2863 

### Describe testing done for PR

Created `prod` plan clusters locally against AWS. Confirmed successful creation of AWS workload cluster.

### Release note

```release-note
AWS classy clusters now populate os-name information from config file and for all possible machineDeployments.
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml [hunk 1]
```diff
@@ -66,13 +66,17 @@ spec:
         metadata:
           annotations:
             #! VVV TODO(vui) os-name handling
```
Comment (anujc25) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:68]: Please remove this TODO as we have addressed it.

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml [hunk 2]
```diff
@@ -66,13 +66,17 @@ spec:
...
+            run.tanzu.vmware.com/resolve-os-image: #@ "ami-region={},os-name={},os-arch={}".format(data.values.AWS_REGION, data.values.OS_NAME, data.values.OS_ARCH)
       #@ if data.values.CLUSTER_PLAN == "prodcc":
       #@overlay/append
       - class: tkg-worker
         name: md-1
         replicas: #@ data.values.WORKER_MACHINE_COUNT_1
         failureDomain: #@ default_az_1
+        metadata:
+          annotations:
+            #! VVV TODO(vui) os-name handling
```
Comment (anujc25) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:78]: Please remove this TODO as we have addressed it.

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml [hunk 3]
```diff
@@ -85,6 +89,10 @@ spec:
         name: md-2
         replicas: #@ data.values.WORKER_MACHINE_COUNT_2
         failureDomain: #@ default_az_2
+        metadata:
+          annotations:
+            #! VVV TODO(vui) os-name handling
```
Comment (anujc25) [inline pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml:94]: Please remove this TODO as we have addressed it.

**Review by anujc25 — APPROVED** [review]
Looks good.

nit: Please remove the TODO before merging the PR.

File: pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml (hunk 1 above)
Comment (tenczar) [reply pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml]: done

**Pull request merged**